katana/native/cases/views.py

```diff
@@ -0,0 +1,370 @@
+	"""
+	path_to_pythonsrc = json.loads(open(path_to_src_python_file).read())['pythonsrcdir'] ;                 
+	jsr = scanfiles.fetch_action_file_names(path_to_pythonsrc,'driver','all');
+	actions = [ os.path.basename(fn)[:-3] for fn in jsr['ProductDrivers']];
+	actions.insert(0,"To_Be_Developed")
+	return JsonResponse({'actions': actions , 'filesinfo' : All_case_action_details })
+
+
+##
+## If we keep this information on file on disk then we'll be left behind 
+## if more actions/keywords are added to the dispalys. 
+## On the other hand getting this information on every call is absurd. 
+## However, the spaghetti code in the py_file_details is too hairy to untangle
+## before the demo and is left alone. 
+##
+def getListOfKeywords(request):
+	if 0: 
+			path_to_src_python_file = navigator.get_katana_dir() + os.sep + "config.json"
+			x= json.loads(open(path_to_src_python_file).read());
+			path_to_pythonsrc = x['pythonsrcdir']; 
+			details = py_file_details(path_to_pythonsrc);
+	else:
+			details = All_case_action_details;
+
+	responseBack = { 'keywords': ["To_Be_Developed",] }
+	driver = request.GET.get('driver');
+		
+	if driver != "To_Be_Developed":
+			
+		print dir(details);
+		print driver
+		print len(details[driver][0])	
+		for item in details[driver][0]: 
+			print item['fn']
+			responseBack['keywords'].append(item['fn']);
+	return JsonResponse(responseBack)
+
+##
+## This is very costly imho. 
+##
+## If we keep this information on file on disk then we'll be left behind 
+## if more actions/keywords are added to the dispalys. 
+## On the other hand getting this information on every call is absurd. 
+## However, the spaghetti code in the py_file_details is too hairy to untangle
+## before the demo and is left alone. 
+##
+def getListOfComments(request):
+	"""
+	Return information about the driver and keyword in the incoming POST request.
+	"""
+	if 0: 
+			path_to_src_python_file = navigator.get_katana_dir() + os.sep + "config.json"
+			x= json.loads(open(path_to_src_python_file).read());
+			path_to_pythonsrc = x['pythonsrcdir']; 
+			details = py_file_details(path_to_pythonsrc);
+	else:
+			details = All_case_action_details;
+			
+	driver  = request.GET.get('driver');
+	keyword = request.GET.get('keyword');
+	responseBack = { 'fields': [] }
+	
+	print "LOOOK::-->",  driver, " ", keyword;
+
+	if driver == "To_Be_Developed" :
+		return JsonResponse(responseBack)
+
+	try:
+		items = details[driver]
+	except:
+		return JsonResponse(responseBack)
+	print "inputs = ", driver,  keyword
+	print "items == ", items;
+	try: 
+		for item in details[driver][0]: 
+			print "fn ==> ", item['fn'], driver, keyword
+			if item['fn'] == keyword: 
+				print item.keys();
+				responseBack['fields'].append(item);
+				break;
+	except:
+		print details[driver]
+	return JsonResponse(responseBack)
+
+def getEmpty():
+	edata={
+		  "Testcase": {
+		    "Details": {
+		      "Name": "set_env_variable",
```

Comment:
Missing results dir, logs dir, expected results.

Reply:
Adding them in to the defaults section when an empty case is created.